x-pack/plugin/relevance-search/src/main/java/org/elasticsearch/xpack/relevancesearch/xsearch/analytics/XSearchAnalyticsService.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.relevancesearch.xsearch.analytics;
+
+import org.elasticsearch.ElasticsearchException;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.DocWriteRequest;
+import org.elasticsearch.action.index.IndexAction;
+import org.elasticsearch.action.index.IndexRequest;
+import org.elasticsearch.action.index.IndexResponse;
+import org.elasticsearch.client.internal.Requests;
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.cluster.metadata.DataStream;
+import org.elasticsearch.cluster.metadata.SearchEngine;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentFactory;
+import org.elasticsearch.xpack.relevancesearch.xsearch.action.XSearchAction;
+
+import java.io.IOException;
+import java.time.ZoneId;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+public class XSearchAnalyticsService {
+
+    private static final Logger logger = LogManager.getLogger(XSearchAnalyticsService.class);
+
+    private final ClusterService clusterService;
+
+    public XSearchAnalyticsService(ClusterService clusterService) {
+        this.clusterService = clusterService;
+    }
+
+    public void recordEvent(XSearchAction.Request request, NodeClient client) {
+        Collection<SearchEngine> engines = getEnginesForRequest(request);
```

Comment:
Not something for now, but I do not think it is a good idea to allow to query several engines in a single _xsearch request.